PdCore/src/main/java/org/puredata/android/io/AudioRecordWrapper.java

```diff
@@ -43,14 +43,17 @@ public AudioRecordWrapper(int sampleRate, int inChannels, int bufferSizePerChann
 			throw new IOException("bad AudioRecord parameters; sr: " + sampleRate + ", ch: " + inChannels + ", bufSize: " + bufferSizePerChannel);
 		}
 		while (recSizeBytes < minRecSizeBytes) recSizeBytes += bufSizeBytes;
-		rec = new AudioRecord(MediaRecorder.AudioSource.MIC, sampleRate, channelConfig, ENCODING, recSizeBytes);
-		if (rec != null && rec.getState() != AudioRecord.STATE_INITIALIZED) {
-			rec.release();
-			throw new IOException("unable to initialize AudioRecord instance for sr: " + sampleRate + ", ch: " + inChannels + ", bufSize: " + bufferSizePerChannel);
-		}
+		try{
+			rec = new AudioRecord(MediaRecorder.AudioSource.MIC, sampleRate, channelConfig, ENCODING, recSizeBytes);
+			if (rec != null && rec.getState() != AudioRecord.STATE_INITIALIZED) {
```

Comment:
```suggestion
			if (rec.getState() != AudioRecord.STATE_INITIALIZED) {
```

There's no way rec can be null here.